cpp/tests/copying/detail_gather_tests.cu

```diff
@@ -57,8 +59,7 @@ TYPED_TEST(GatherTest, GatherDetailDeviceVectorTest)
 
   // test with device vector iterators
   {
-    std::unique_ptr<cudf::table> result =
-      cudf::detail::gather(source_table, gather_map.begin(), gather_map.end());
+    std::unique_ptr<cudf::table> result = cudf::gather(source_table, gather_map->view());
```

Comment:
This is an interesting change. 
The intention of this source file seems to be wanting to test `cudf::detail::gather` functions -- specifically the iterator interfaces as per this comment from above:
https://github.com/rapidsai/cudf/pull/12888/files#diff-88fa573b86cc7871ca7ce24babaa1651cb41a2c76de8fdab1925c21dae450fc4R44-R45
Perhaps these tests are really not needed and this source file itself is not necessary at all.

Reply:
Hmm yeah that is a good point. This file had two invocations of `detail::gather` and it looks like the other one was properly passing the stream through. I think the minimal change for this PR would be to just pass the stream, then to consider removing in a separate PR, WDYT? I made the change that I did thinking that it would be best to move away from using detail APIs in test altogether, but I also want to limit scope creep.

Reply:
We can evaluate removing these tests in a separate PR.
But then I think you should change this back to calling the `detail` API right?

Reply:
Yup was waiting for your approval on the plan before pushing the changes. I'll update it now.